source/mir/math/func/powi.d

```diff
@@ -0,0 +1,72 @@
+/++
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: John Michael Hall
+
+Copyright: 2020 Mir Stat Authors.
++/
+
+module mir.math.func.powi;
```

Comment:
Why not http://mir-core.libmir.org/mir_math_common.html#.powi ?

Reply:
We should convert the first argument to double if it an integer.

Reply:
@9il The overloads I added are covering a few cases that aren't currently handled by what is in `mir.core`. For instance, the first is casting away const-ness. The second does what you are saying and will cast the first argument to double if it is implicitly convertible to double. The third handles unsigned integers for the power (casting to int). The fourth handles the interaction of the second and third, and the fifth is a final fallback (for instance for handling complex numbers). Four out of five are just calling `mir.core`'s implementation after casts. It's just easier for the `moment` code to not have to deal with these itself.

I'm happy to add the overloads to `mir.core` instead of here.